app/models/views/application_result.rb

```diff
@@ -0,0 +1,48 @@
+module Views
+  class ApplicationResult
+
+    def initialize(application)
+      @application = application
+    end
+
+    def result
+      %w[full part none].include?(outcome) ? outcome : 'error'
+    end
+
+    def amount_to_pay
+      if evidence_or_application.amount_to_pay.present?
+        "£#{evidence_or_application.amount_to_pay.round}"
+      end
+    end
+
+    def savings
+      result_translations[@application.savings_investment_valid?.to_s]
+    end
+
+    def income
+      result_translations[%w[full part].include?(result).to_s]
+    end
+
+    private
+
+    def outcome
+      case evidence_or_application
+      when EvidenceCheck
+        evidence_or_application.outcome
+      when Application
+        evidence_or_application.application_outcome
+      end
+    end
```

Comment:
@novotnyjakub you probably have this better in your head right now as you were knee deep in this code, so you might be able to answer this:

Could we potentially use duck typing here? Instead of doing this check to see if it's **EvidenceCheck** or **Application**, can't we just pass in an object instance, that answers to some method name that would produce the desired result e.g. @object.outcome ?

Or is this maybe too complex now to derive because we don't have the **Decision** model which could be queried?
Something like ```@application.decision.outcome```, which would be updated regardless if there was or wasn't an evidence check?

What are your thoughts?